### ISSUE TITLE
SearchService does not handle lucene ParseException

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#326]
This is more like a `CompassSource` thing to handle. There should be checked exceptions for malformed `SearchSettings` parameters.

```
SearchEngineQueryParseException: Failed to parse query [and \]; nested exception is org.apache.lucene.queryParser.ParseException: Cannot parse 'and \': Lexical error at line 1, column 6.  Encountered: <EOF> after : ""
org.compass.core.engine.SearchEngineQueryParseException: Failed to parse query [and \]; nested exception is org.apache.lucene.queryParser.ParseException: Cannot parse 'and \': Lexical error at line 1, column 6.  Encountered: <EOF> after : ""
org.apache.lucene.queryParser.ParseException: Cannot parse 'and \': Lexical error at line 1, column 6.  Encountered: <EOF> after : ""
    at org.apache.lucene.queryParser.QueryParser.parse(QueryParser.java:216)
    at org.compass.core.lucene.engine.queryparser.DefaultLuceneQueryParser.parse(DefaultLuceneQueryParser.java:111)
    at org.compass.core.lucene.engine.query.LuceneSearchEngineQueryStringBuilder.toQuery(LuceneSearchEngineQueryStringBuilder.java:93)
```

**Comment by arteymix** [comment]
I'm adding a checked `SearchException` so that this can be safely handled by the frontend and other location that rely on the search service.

**Comment by arteymix** [comment]
It should also make the REST API more robust. I'll add a `400 Bad Request` exception for malformed search query.

**Comment by arteymix** [comment]
Fixed in c94f805f556eb2b3c8a514e120040fff6c19d631!
